### ISSUE TITLE
Self referencing loop detected

### ISSUE THREAD

**Issue by secana** [opentracing/opentracing-csharp#102]
I've found a possible bug in

**Jaeger v0.0.10
OpenTracing v0.12.0**

Code example:
```
using (var scope = tracer
                    .BuildSpan("spanName")
                    .StartActive(true))
                {
                   // Do stff
                }
```

Throws:
```
Newtonsoft.Json.JsonSerializationException
  HResult=0x80131500
  Message=Self referencing loop detected for property 'Span' with type 'Jaeger.Span'. Path 'Tracer.ScopeManager.Active'.
  Source=Newtonsoft.Json
  StackTrace:
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.CheckForCircularReference(JsonWriter writer, Object value, JsonProperty property, JsonContract contract, JsonContainerContract containerContract, JsonProperty containerProperty)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.CalculatePropertyValues(JsonWriter writer, Object value, JsonContainerContract contract, JsonProperty member, JsonProperty property, JsonContract& memberContract, Object& memberValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeObject(JsonWriter writer, Object value, JsonObjectContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeObject(JsonWriter writer, Object value, JsonObjectContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeObject(JsonWriter writer, Object value, JsonObjectContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeObject(JsonWriter writer, Object value, JsonObjectContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.Serialize(JsonWriter jsonWriter, Object value, Type objectType)
   at Newtonsoft.Json.JsonSerializer.SerializeInternal(JsonWriter jsonWriter, Object value, Type objectType)
   at Newtonsoft.Json.JsonConvert.SerializeObjectInternal(Object value, Type type, JsonSerializer jsonSerializer)
   at Jaeger.Reporters.LoggingReporter.Report(Span span)
   at Jaeger.Reporters.CompositeReporter.Report(Span span)
   at Jaeger.Tracer.ReportSpan(Span span)
   at Jaeger.Span.FinishInternal(DateTime finishTimestampUtc)
   at OpenTracing.Util.AsyncLocalScope.Dispose()
   at Client.ClientProgram.Main(String[] args) 
```
On dispose (Span.Finish())

If the tracer is from interested here is the code I used to create it:

```
public static Jaeger.Tracer
            Build(string serviceName, string jaegerHost = "localhost",
                int jaegerPort = 6831)
        {
            var sender        = new UdpSender(jaegerHost, jaegerPort, 1000);
            var loggerFactory = new LoggerFactory().AddConsole();
            var remoteReporter = new RemoteReporter.Builder()
                .WithLoggerFactory(loggerFactory)
                .WithSender(sender)
                .Build();

            var loggingReporter = new LoggingReporter(loggerFactory);
            var sampler         = new ConstSampler(true);
            var tracer = new Jaeger.Tracer.Builder(serviceName)
                .WithLoggerFactory(loggerFactory)
                .WithReporter(new CompositeReporter(loggingReporter, remoteReporter))
                .WithSampler(sampler)
                .Build();

            return tracer;
        }
```

Any help is appreciate!

Regards, 
 Secana

**Comment by cwe1ss** [comment]
Hi @secana. That's a bug in Jaeger - we're currently working on it: https://github.com/jaegertracing/jaeger-client-csharp/issues/68

**Comment by secana** [comment]
Oh, sry for the duplicate. Seems that I've searched in the wrong repository for an issue. If the issue will be fixed in jaegertracing/jaeger-client-csharp#68 this issue can be closed.

**Comment by Falco20019** [comment]
Hi @secana. It‘s fixed in the code and will be part of [v0.1.0](https://github.com/jaegertracing/jaeger-client-csharp/milestone/1) which should release today or tomorrow.
